mlir/lib/Dialect/Bufferization/Transforms/Bufferize.cpp

```diff
@@ -67,10 +67,14 @@ BufferizeTypeConverter::BufferizeTypeConverter() {
                               ValueRange inputs, Location loc) -> Value {
     assert(inputs.size() == 1 && "expected exactly one input");
 
+    // Unranked to ranked casts must be explicit.
+    if (auto inputType = dyn_cast<UnrankedMemRefType>(inputs[0].getType()))
```

Comment:
Why is this needed?

Reply:
I removed this. At the end of this materialization function there is ` llvm_unreachable("only tensor/memref input types supported");`, but if the input is `UnrankedMemRefType`, then it will still hit the `unreachable`.  Is that by design?